tests/functional/metadata/mongodb/getObject.spec.js

```diff
@@ -0,0 +1,263 @@
+const async = require('async');
+const assert = require('assert');
+const werelogs = require('werelogs');
+const { MongoMemoryReplSet } = require('mongodb-memory-server');
+const { errors, versioning } = require('../../../../index');
+const logger = new werelogs.Logger('MongoClientInterface', 'debug', 'debug');
+const BucketInfo = require('../../../../lib/models/BucketInfo');
+const MetadataWrapper =
+    require('../../../../lib/storage/metadata/MetadataWrapper');
+const genVID = versioning.VersionID.generateVersionId;
+const { DbPrefixes } = versioning.VersioningConstants;
+
+const IMPL_NAME = 'mongodb';
+const DB_NAME = 'metadata';
+const BUCKET_NAME = 'test-bucket';
+const replicationGroupId = 'RG001';
+
+const mongoserver = new MongoMemoryReplSet({
+    debug: false,
+    instanceOpts: [
+        { port: 27019 },
+    ],
+    replSet: {
+        name: 'rs0',
+        count: 1,
+        DB_NAME,
+        storageEngine: 'ephemeralForTest',
+    },
+});
+
+let uidCounter = 0;
+function generateVersionId() {
+    return genVID(`${process.pid}.${uidCounter++}`,
+                    replicationGroupId);
+}
+
+function formatMasterKeyV0(key) {
+    return `${key}`;
+}
+
+function formatMasterKeyV1(key) {
+    return `${DbPrefixes.Master}${key}`;
+}
+
+function formatMasterKey(key, vFormat) {
+    if (vFormat === 'v0') {
+        return formatMasterKeyV0(key);
+    }
+    return formatMasterKeyV1(key);
+}
+
+describe('MongoClientInterface::metadata.getObjectMD', () => {
+    let metadata;
+    let collection;
+    let versionId1;
+
+    const params = {
+        objName: 'pfx1-test-object',
+        objVal: {
+            key: 'pfx1-test-object',
+            versionId: 'null',
+        },
+        vFormat: 'v0',
+    };
+
+    function updateMasterObject(objName, versionId, objVal, vFormat, cb) {
+        const mKey = formatMasterKey(objName, vFormat);
+        collection.updateOne(
+            {
+                _id: mKey,
+                $or: [{
+                    'value.versionId': {
+                        $exists: false,
+                    },
+                },
+                {
+                    'value.versionId': {
+                        $gt: versionId,
+                    },
+                },
+                        ],
+            },
+            {
+                $set: { _id: objName, value: objVal },
+            },
+            { upsert: true },
+            err => {
+                if (err) {
```

Comment:
this is copy-paste of code we don't need (not creating 2 keys at the same time?), and most likely don't want (this is  test, not prod code) : keep things simple.

Reply:
fixed in https://github.com/scality/Arsenal/pull/1729/commits/a6f48b9f342344e6df5ca2e3b7a2d03d8737fbba